server/server.js

```diff
@@ -1,15 +1,83 @@
+require("dotenv").config();
 const express = require("express");
 const app = express();
 const port = process.env.PORT || 5000;
+const cors = require("cors");
+app.use(cors());
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+const videos = require("./exampleresponse.json");
 
-// Store and retrieve your videos from here
-// If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+app.use(express.json());
 
 // GET "/"
-app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
-  res.send({ express: "Your Backend Service is Running" });
+app.get("/videos", (req, res) => {
+  res.status(200).json(videos);
+});
+
+app.post("/videos", (req, res) => {
+  const { title, url } = req.body;
+  if (!title || !url || !url.startsWith("https://www.youtube.com")) {
+    res.status(400).json({
+      result: "failure",
+      message: "Video could not be saved",
+    });
+  } else {
+    const idList = videos.map((video) => video.id);
+    const id = Math.max(...idList) + 1;
+    const newVideo = {
+      id,
+      title,
+      url,
+      rating: 0,
+    };
+    videos.push(newVideo);
+    res.status(201).json({ id });
+  }
+});
+
+app.get("/videos/:id", (req, res) => {
+  const id = Number(req.params.id);
+  const matchingVideo = videos.find((video) => {
+    return video.id === id;
+  });
+  if (!matchingVideo) {
+    res.status(400).send("No matching video with this ID exists.");
+  } else {
+    res.status(200).json({ matchingVideo });
+  }
+});
+
+app.delete("/videos/:id", (req, res) => {
+  let id = Number(req.params.id);
+  const matchingVideo = videos.find((video) => {
+    return video.id === id;
+  });
+
+  if (!matchingVideo) {
+    res.status(400).json({
+      result: "failure",
+      message: "Video could not be deleted",
+      id: id,
+    });
+  } else {
+    const videoIndexToBeDeleted = videos.indexOf(matchingVideo);
+    videos.splice(videoIndexToBeDeleted, 1);
+    res.status(200).json({});
+  }
+});
+
+app.put("/videos/:id", (req, res) => {
+  const newVideo = req.body;
+  let id = Number(req.params.id);
+  const videoIndex = videos.findIndex((video) => {
```

Comment:
In the previous function you used `videos.indexOf`, and here you're using `videos.findIndex` - is there a difference? Why did you choose one and the other?